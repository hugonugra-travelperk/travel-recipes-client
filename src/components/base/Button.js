import styled from "styled-components";

export const Button = styled.button`
  /* Adapt the colors based on primary prop */
  background: ${props => props.primary ? "blue" : "white"};
  color: ${props => props.primary ? "white" : "palevioletred"};
  font-size: 1.25em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid blue;
  border-radius: 3px;
  &:hover {
    background: #ffc726;
  }
  &:active {
    background: lightblue;
  }
`;
