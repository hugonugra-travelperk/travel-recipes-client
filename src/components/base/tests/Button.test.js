import {render, cleanup} from "@testing-library/react"
import "@testing-library/jest-dom";
import {Button} from "../Button"

it("renders", () => {
  const {asFragment} = render(<Button>Test text</Button>);
  expect(asFragment()).toMatchSnapshot();
});
