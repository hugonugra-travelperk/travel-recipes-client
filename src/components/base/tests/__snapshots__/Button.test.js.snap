// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL gDoRyd"
  >
    Test text
  </button>
</DocumentFragment>
`;
